drivers/hmis/app/models/hmis/hud/concerns/with_strict_attributes.rb

```diff
@@ -0,0 +1,24 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Hmis::Hud::Concerns::WithStrictAttributes
+  extend ActiveSupport::Concern
+
+  included do
+    raise "#{self} cannot include Hmis::Hud::Concerns::WithStrictAttributes because the table #{table_name} does not exist. Try moving `self.table_name` assignment to the top of the class." unless table_exists?
+
+    columns.each do |column|
+      case column.type
+      when :integer
+        attribute column.name, Hmis::StrictInteger.new
+      when :decimal
+        attribute column.name, Hmis::StrictDecimal.new
+      end
```

Comment:
We should handle big ints and floats here too per Therons comment. It looks like CustomService FAAmount is a float for some reason (not ideal that thats different from Service.FAAmount, dont think it was intentional)

I think the possible values for type are the values from `NATIVE_DATABASE_TYPES ` here https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/PostgreSQLAdapter.html

Reply:
Interesting, it looks like there is a `money` type supported as well, but I think we don't use it?

Reply:
Not that I know of